backend/src/packages/chaiNNer_pytorch/pytorch/restoration/upscale_face.py

```diff
@@ -120,6 +121,24 @@ def upscale(
         NumberInput(
             label="Output Scale", default=8, minimum=1, maximum=8, unit="x"
         ).with_id(3),
+        if_group(
+            Condition.type(0, 'PyTorchModel { arch: "CodeFormer" }')
+            & (
+                Condition.type(
+                    0, 'PyTorchModel { arch: "GFPGAN" | "RestoreFormer" }'
+                ).__invert__()
+            )
+        )(
```

Comment:
We talked about this on discord. But I thought about this a bit more, and negating type conditions is actually incorrect in general. We can't really enforce this, but I'll make an issue about this for now.
```suggestion
        if_group(Condition.type(0, 'PyTorchModel { arch: "CodeFormer" }'))(
```

Reply:
I don't want this showing up unless codeformer is connected. It's not even useful to have this being conditional without that. If it's gonna show up without anything connected to it i might as well just make it show up all the time, because it goes from "oh there's a different option when i connect codeformer" to "hey where did that option go when i connected gfpgan wtf"

Reply:
Actually fine i give up on this. Let's do it your way and see what happens

Reply:
> If it's gonna show up without anything connected to it

It won't show up unless *something* is connected, don't worry. We had this discussion before when we first added type conditions :)

Reply:
That's not how it was functioning last night, it was showing up all the time

Reply:
You're right... wtf. I'll look into it. This is not how they are supposed to behave.